### PR TITLE
add feature: query top n process instance order by running duration

### PR DESCRIPTION
**Pull request by RedemptionC**
## What is the purpose of the pull request

complete feature(#3191)

## Brief change log
* add function in ProcessInstanceController.java , ProcessInstanceService.java  , ProcessInstanceMapper.java 
* add some test in ProcessInstanceServiceTest.java , ProcessInstanceMapperTest.java 
* add some constant/status in Status.java ,  Constants.java 
* add some sql in ProcessInstanceMapper.xml

## Verify this pull request

* Manually verified the change by testing locally.

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java [hunk 1]
```diff
@@ -204,6 +204,28 @@ public Result queryProcessInstanceById(@ApiIgnore @RequestAttribute(value = Cons
...
+    @ApiException(QUERY_PROCESS_INSTANCE_BY_ID_ERROR)
+    public Result queryTopNLongestRunningProcessInstance(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
+                                                         @ApiParam(name = "projectName", value = "PROJECT_NAME", required = true) @PathVariable String projectName,
+                                                         @RequestParam("size") Integer size,
+                                                         @RequestParam(value = "startTime",required = true) String startTime,
+                                                         @RequestParam(value = "endTime",required = true) String endTime
+
+    ){
+        logger.info("query top {} SUCCESS process instance order by running time which started between startTime and endTime ,login user:{},project name:{}",size,
+                loginUser.getUserName(),projectName);
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java:224]: good job, but I don't think this log is meaningful

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java [hunk 2]
```diff
@@ -98,6 +98,50 @@
...
+        if (Objects.isNull(endTime)) {
+            putMsg(result, Status.DATA_IS_NULL, Constants.END_TIME);
+            return result;
+        }
+        // TODO the format really matters
+        Date end=DateUtils.stringToDate(endTime);
+        if (start.getTime() > end.getTime()) {
+            putMsg(result, Status.START_TIME_BIGGER_THAN_END_TIME_ERROR, startTime, endTime);
+            return result;
+        }
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java:138]: The verification should be advanced

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java (hunk 1 above)
Comment (RedemptionC) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java:224]: can you give me some advice?It seems other method log the same way

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java (hunk 2 above)
Comment (RedemptionC) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java:138]: I don't understand...
and, I wonder why some unit test cause error while I never modify them?

Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java:138]: Test failure is not your problem, we will fix it later, if you are interested, you can also try to solve it

Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java:138]: > I don't understand...
> and, I wonder why some unit test cause error while I never modify them?
1:These parameter verification should be judged before SQL execution.
2:Single test failure is not your problem, we will fix it later, if you are interested, you can also try to solve it

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java (hunk 1 above)
Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java:224]: In this case,  this kind of log does not make much sense, on the other hand, it makes our log files huge.

Comment (RedemptionC) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessInstanceController.java:224]: could you please 

> In this case, this kind of log does not make much sense, on the other hand, it makes our log files huge.

could you please give me some advice on how to make the log meaningful?
I have found that I should replace the starttime and endtime with params

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java (hunk 2 above)
Comment (RedemptionC) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/ProcessInstanceService.java:138]: which sql ? I did put the queryTopNProcessInstance query after the params verification

**Pull request closed without merging**